st2common/st2common/openapi.yaml.j2

```diff
@@ -973,6 +973,11 @@ paths:
           description: Show secrets in plain text
           type: boolean
       x-parameters:
+        - name: filter
+          in: query
+          x-as: advanced_filters
+          description: Advanced filter string. Overrides other filters.
```

Comment:
Also, any particular reason this is a string not an object / array?

Reply:
Because it is much harder to express complex query logic in array or an object. Both require use of structural characters such as `{`, `[` or `,` that only create clutter and don't add anything useful to the query.

Because it being a GET request requires us to also make `filter` a query parameter which by itself is a string.

Because it helps us to keep all the logic in one place (in st2api) and avoid additional changes if we decide to change the syntax (and we may).

Reply:
I'm fine in this specific case, but I would argue that the input type doesn't matter that much.

That's why we have openapi for - to also cast and convert values as needed (same goes for boolean arguments, etc.)